resolves the type mismatch between the internel ioctl type and the library one by casting to a void pointer
this is a BAD solution and effectively just forces ignoring the issue, as in this specific circumstance
there shouldn't be an issue doing that.
however it is not stable or portable, and shouldn't be used longterm.
diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index 9736cd5..4fc53f9 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -544,7 +544,7 @@ gst_v4l2_object_new (GstElement * element,
     v4l2object->fd_open = NULL;
     v4l2object->close = close;
     v4l2object->dup = dup;
-    v4l2object->ioctl = ioctl;
+    v4l2object->ioctl = (void *)ioctl;
     v4l2object->read = read;
     v4l2object->mmap = mmap;
     v4l2object->munmap = munmap;
